test/home_bloc_test.dart

```diff
@@ -0,0 +1,152 @@
+import 'mock_repository.dart';
+
+void main() {
+  group('HomeBlocTest', () {
+    late HomeBloc homeBloc;
+    MockRepository mockRepository = MockRepository();
+    GetCharactersUseCase getCharactersUseCase;
+
+    setUp(() {
+      EquatableConfig.stringify = true;
+      getCharactersUseCase = GetCharactersUseCase(repository: mockRepository);
+      homeBloc = HomeBloc(getCharactersUseCase: getCharactersUseCase);
+    });
+
+    blocTest<HomeBloc, HomeState>(
+      'First Screen launch when list of heroes is empty',
+      build: () => homeBloc,
+      act: (homeBloc) => homeBloc.add(ReadyForData()),
+      wait: const Duration(milliseconds: 300),
+      expect: () => [
+        HomeState(
+          loading: Loading.fullScreen,
+          hasReachedMax: false,
+          characters: [],
+          error: null,
+        ),
+        HomeState(
+          loading: Loading.loaded,
+          hasReachedMax: false,
+          characters: mockCharactersViewData.getRange(0, 11).toList(),
+          error: null,
+        ),
+      ],
+    );
+
+    blocTest<HomeBloc, HomeState>(
+      'When list scrolled to bottom loading is shown and new page is requested',
+      build: () => homeBloc,
+      act: (homeBloc) => homeBloc.add(ScrolledToEnd()),
+      wait: const Duration(milliseconds: 300),
+      seed: () => HomeState(
+        loading: Loading.loaded,
+        hasReachedMax: false,
+        characters: mockCharactersViewData.getRange(0, 11).toList(),
+        error: null,
+      ),
+      expect: () => [
+        HomeState(
+          loading: Loading.nextPage,
+          hasReachedMax: false,
+          characters: mockCharactersViewData.getRange(0, 11).toList(),
+          error: null,
+        ),
+        HomeState(
+          loading: Loading.loaded,
+          hasReachedMax: false,
+          characters: mockCharactersViewData.getRange(0, 22).toList(),
+          error: null,
+        ),
+      ],
+    );
+
+    blocTest<HomeBloc, HomeState>(
+      'When all heroes is downloaded and requested lust page ',
+      build: () => homeBloc,
+      act: (homeBloc) => homeBloc.add(ScrolledToEnd()),
+      wait: const Duration(milliseconds: 300),
+      seed: () => HomeState(
+        loading: Loading.loaded,
+        hasReachedMax: false,
+        characters: mockCharactersViewData.getRange(0, 22).toList(),
+        error: null,
+      ),
+      expect: () => [
+        HomeState(
+          loading: Loading.nextPage,
+          hasReachedMax: false,
+          characters: mockCharactersViewData.getRange(0, 22).toList(),
+          error: null,
+        ),
+        HomeState(
+          loading: Loading.loaded,
+          hasReachedMax: false,
+          characters: mockCharactersViewData.getRange(0, 30).toList(),
+          error: null,
+        ),
+      ],
+    );
+
+    blocTest<HomeBloc, HomeState>(
+      'When lust page is downloaded and user scrolled to end again',
```

Comment:
After "when" must go "then", so "when last page is downloaded then something is happen"